toolkit/resources/assets/efi/grub/grub.cfg

```diff
@@ -1,2 +1,6 @@
 search -n -u {{.BootUUID}} -s
-configfile /boot/grub2/grub.cfg
\ No newline at end of file
+
+# If '/boot' is a seperate partition, BootUUID will point directly to '/boot'.
+# In this case we should omit the '/boot' prefix from all paths.
+set bootprefix={{.BootPrefix}}
+configfile $bootprefix/grub2/grub.cfg
```

Comment:
>$bootprefix/grub2/grub.cfg [](start = 11, length = 26)

So what are the options for the path to the config file now? I understand we cannot have the contents of `/boot` on the same partition as the rest of the root, since now this root partition is hashed/signed, but why would its path change?

Reply:
Grub runs relative to its current partition, so if we mount `/boot` as a separate partition the relative path will be inside the `/boot` folder already. Ie Normally grub loads `(hd0,1)/boot/grub2/grub.cfg`, but now we are loading `(hd0,2)` which IS /boot, so instead we need to load `(hd0,2)/grub2/grub.cfg`.

Reply:
Minor note - grub's root variable is set in the `search -n -u {{.BootUUID}} -s` command. Specifically the -s option writes the value to a variable. Default variable is "root"